pkg/local-gadget-manager/local-gadget-manager_test.go

```diff
@@ -159,6 +161,26 @@ func checkFdList(t *testing.T, initialFdList string, attempts int, sleep time.Du
 	}
 }
 
+// TestClose tests that resources aren't leaked after calling Close()
+func TestClose(t *testing.T) {
+	if !*rootTest {
+		t.Skip("skipping test requiring root.")
+	}
+
+	initialFdList := currentFdList(t)
+
+	localGadgetManager, err := NewManager([]*containerutils.RuntimeConfig{{Name: "docker"}})
+	if err != nil {
+		t.Fatalf("Failed to start local gadget manager: %s", err)
+	}
+
+	localGadgetManager.Close()
+
+	// We need a quite long try period here because the runc
+	// notifier takes a bit long to release all resources
```

Comment:
How long does it take? It is because a file descriptor is closed when the GC releases a os.File? If so, that would be a bug.

Reply:
That comment is actually outdated as this long period is not needed anymore. It was fixed by your suggestion to the `fanotifyFlags` and by some closing logic added to `func (n *RuncNotifier) Close()`.

Reply:
Comment removed and timeout updated to be the same as in other tests.